src/panel/direction/DirectionPanel.jsx

```diff
@@ -80,6 +83,35 @@ export default class DirectionPanel extends React.Component {
     document.body.classList.remove('directions-open');
   }
 
+  getAddresses = (origin, destination) => {
+    const inputTextPromises = [
+      origin
```

Comment:
We should find a way to avoid these unclear nested ternaries.
For example:
```js
origin && origin.type === 'latlon' ? IdunnPoi.poiApiLoad(origin) : origin
```
And sorry about the `Promise.resolve` that I used elsewhere, you can remove it if you want.
I thought it was needed but it can be omitted with `Promise.all`, any non-Promise value is considered as a Promise resolving with this value.